plugins/catalog-backend/src/next/DefaultCatalogProcessingEngine.ts

```diff
@@ -167,8 +167,7 @@ export class DefaultCatalogProcessingEngine implements CatalogProcessingEngine {
             hashBuilder = hashBuilder
               .update(stableStringify({ ...result.completedEntity }))
               .update(stableStringify([...result.deferredEntities]))
-              .update(stableStringify([...result.relations]))
-              .update(stableStringify(Object.fromEntries(result.state)));
+              .update(stableStringify([...result.relations]));
```

Comment:
i was expecting some code here that actually persisted the new state if it was changed?

Reply:
4. Update the cache only if it has been written to

As I've said before, I'm unsure about the whole rewrite-always strategy, which goes together with the 4. suggestion here

Reply:
And also this thinking aligns with what we said online earlier - what if the caching concern was completely orthogonal to all of this (e.g. like it would be if it were backed by memcache/redis)? Then the code in here would not care at all about what the state ended up doing within its own concerns

Reply:
4 isn't really related to this though, as a write can either result in a different cache value or the same, while no write would always result in the same. What this logic is dealing with is what to do when the output and/or cache data has changed, and we'd still need to choose either 1, 2, or 3.

I think we could consider implementing 4 further down in the interface used by the processors though, to potentially get around the need for the extra `set()`

Reply:
Discussed this all and settled for option 2, along with separating out the writing of the state to a separate DB call